misc/python/materialize/cli/mzcompose.py

```diff
@@ -103,8 +103,20 @@ def main(argv: List[str]) -> int:
 
     # The `run` command requires special handling.
     if args.command == "run":
-        workflow = composition.workflows.get(args.first_command_arg, None)
-        if workflow is None:
+        try:
+            workflow = composition.get_workflow(
+                dict(os.environ), args.first_command_arg
+            )
+            # The user has specified a workflow rather than a service. Run the
+            # workflow instead of Docker Compose.
+            if args.remainder:
+                raise errors.MzRuntimeError(
+                    f"cannot specify extra arguments ({' '.join(args.remainder)}) "
+                    "when specifying a workflow (rather than a container)"
+                )
+            workflow.run()
+            return 0
+        except KeyError:
```

Comment:
the downside of this refactoring is that it will swallow keyerrors in `workflow.run`, I'd rather move the `workflow.run` outside of the try/catch or into a new `else:` block

Reply:
Okay, can do!

Reply:
I put this into an `else` block.